libs/wingsdk/src/cloud/table.ts

```diff
@@ -0,0 +1,134 @@
+import { Construct } from "constructs";
+import { fqnForType } from "../constants";
+import { App, Resource } from "../core";
+import { Json } from "../std";
+
+/**
+ * Global identifier for `Table`.
+ */
+export const TABLE_FQN = fqnForType("cloud.Table");
+
+/**
+ * Table column types
+ */
+export enum ColumnType {
+  /** string type */
+  STRING,
+  /** number type */
+  NUMBER,
+  /** bool type */
+  BOOLEAN,
+  /** date type */
+  DATE,
+  /** json type */
+  JSON,
+}
+
+/**
+ * Properties for `Table`.
+ */
+export interface TableProps {
+  /**
+   * The table's columns.
+   * @default undefined
+   */
+  readonly columns: { [key: string]: ColumnType };
+  /**
+   * The table's primary key. No two rows can have the same value for the
+   * primary key.
+   * @default undefined
+   */
+  readonly primaryKey: string;
+}
+
+/**
+ * Represents a NoSQL database table that can be used to store and query data.
+ * @inflight `@winglang/sdk.cloud.ITableClient`
+ */
+export abstract class Table extends Resource {
+  /**
+   * Create a new `Table` instance.
+   * @internal
+   */
+  public static _newTable(
+    scope: Construct,
+    id: string,
+    props: TableProps
+  ): Table {
+    return App.of(scope).newAbstract(TABLE_FQN, scope, id, props);
+  }
+
+  /**
+   * Table primary key name
+   */
+  public readonly primaryKey: string;
+  /**
+   * Table columns
+   */
+  public readonly columns: { [key: string]: ColumnType };
+  public readonly stateful = true;
+  constructor(scope: Construct, id: string, props: TableProps) {
+    super(scope, id);
+
+    this.display.title = "Table";
+    this.display.description =
+      "A cloud NoSQL database table that can be used to store and query data";
+
+    this.primaryKey = props.primaryKey;
+    this.columns = props.columns;
+  }
+}
+
+/**
+ * Inflight interface for `Table`.
+ */
+export interface ITableClient {
+  /**
+   * Insert a row into the table.
+   * @param row data to be inserted.
+   * @inflight
+   */
+  insert(row: Json): void;
```

Comment:
Curious why row is Json type here. In the spec we have `insert(row: Map<str, Json>)` which I assume is supposed to be a column to Json content mapping. IE.
```ts
let row = {
  "id": Json "some_id",
  "headers": Json {
    some_header: "value",
    numbers: [1, 2, 3]
  }
};

table.insert(row);
```

Maybe the DX is nicer with just Json, but wanted to call out the spec difference.

Reply:
Change to Map give this error

```bash
src/cloud/table.ts:111:10 - error JSII9002: Unable to resolve type "Map". It may be @internal or not exported from the module's entry point (as configured in "package.json" as "main").

       111   insert(row: Map<string, Json>): void;
                    ~~~~~~~~~~~~~~~~~~~~~~
       src/cloud/table.ts:111:15 - error JSII3001: Type "Map" cannot be used as the parameter type because it is private or @internal

       111   insert(row: Map<string, Json>): void;
                         ~~~~~~~~~~~~~~~~~

         node_modules/jsii/node_modules/typescript/lib/lib.es2015.collection.d.ts:36:13
           36 declare var Map: MapConstructor;
                          ~~~~~~~~~~~~~~~~~~~
           The referenced type is declared here
```